### ISSUE TITLE
backpropgation on chunks?

### ISSUE THREAD

**Issue by vr25** [AndriyMulyar/bert_document_classification#14]
Hi,

When the document chunks are fed to the data parallel model, how is the loss backpropagated? Is it for every chunk?

Also, do you unfreeze and fine-tune for the classification task?

Thank you!

**Comment by AndriyMulyar** [comment]
1. Yes, separate for every chunk.
2. In our datasets we found it sufficient to fine-tune only the final
transformer layer.

**Comment by vr25** [comment]
Also, what is the maximum number of chunks in the entire dataset?

The default config has bert_batch_size=7 but I have some documents with a total number of chunks=125 per document. In such cases, if set bert_batch_size to 125, I run into CUDA OOM error.

Any suggestions for this?

Thanks!
